src/cls/CMakeLists.txt

```diff
@@ -135,3 +135,17 @@ endif (WITH_CEPHFS)
 
 add_library(cls_numops_client STATIC numops/cls_numops_client.cc)
 
+add_library(cls_lua SHARED
+    lua/cls_lua.cc
+    lua/lua_bufferlist.cc
+)
+set_target_properties(cls_lua PROPERTIES VERSION "1.0.0" SOVERSION "1")
+install(TARGETS cls_lua DESTINATION ${cls_dir})
+target_link_libraries(cls_lua
+    liblua
```

Comment:
from https://github.com/ceph/lua/blob/lua-5.1/CMakeLists.txt, the project actually defines a 'liblua' target that we refer to here:
```cmake
add_library ( liblua SHARED ${SRC_LIB} ${LUA_DLL_RC} ${LUA_DEF} )
target_link_libraries ( liblua ${LIBS} )
set_target_properties ( liblua PROPERTIES OUTPUT_NAME lua CLEAN_DIRECT_OUTPUT 1 )
```
(the `OUTPUT_NAME lua` property is what prevents it from naming the output file libliblua.so)